### PR TITLE
Add the `HellipButton` component and update the `HiddenTextExpender`

### PR DESCRIPTION
**Pull request by Artemisia-Absinthium**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -46,6 +62,14 @@ The category for changes related to documentation, testing and tooling. Also, fo
 
     *Manuel Puyol*
 
+* Fix components not rendering in Storybook because of kebab case arguments.
```
Comment (Artemisia-Absinthium) [inline CHANGELOG.md:65]: I made those two small changes as I encounter them during the creation of that PR.

Is it OK to add it to that PR or should I make separates ones?

File: app/components/primer/hellip_button.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        "inline" => inline
+      )
+    end
+
+    def call
+      render(Primer::BaseButton.new(**@system_arguments)) { "&hellip;".html_safe }
+    end
+
+    def validate!
+      validate_aria_label
```
Comment (joelhawksley) [inline app/components/primer/hellip_button.rb:40]: Why not just inline this on 25?

**Review by joelhawksley — COMMENTED** [review]
@owenniblock I'm curious: is there other place we think we'll use this pattern besides the HiddenTextExpander?

**Comment by owenniblock** [comment]
@joelhawksley - I have no idea sorry! I'm picking up this issue https://github.com/primer/view_components/issues/637 and trying to get this PR ready to merge.

I haven't got all the context here but the Issue suggests that there's a desire to split the behavioural component away from the button itself?

Perhaps the `HiddenTextExpander` could be extended to allow other button types in future - not just the `HellipButton`?

File: app/components/primer/hellip_button.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Primer
+  # Use `HellipButton` to render a button with an hellip. Often used for hidden text expanders.
```
Comment (camertron) [inline app/components/primer/hellip_button.rb:4]:
```suggestion
  # Use `HellipButton` to render a button with a hellip. Often used for hidden text expanders.
```

File: app/components/primer/hellip_button.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  # @accessibility
+  #   Always set an accessible label to help the user interact with the component.
+  #
+  #   * This button is displaying an hellip as its content (The three dots character). Therefore a label is needed for screen readers.
```
Comment (camertron) [inline app/components/primer/hellip_button.rb:8]:
```suggestion
  #   * This button is displaying a hellip as its content (The three dots character). Therefore a label is needed for screen readers.
```

**Review by camertron — COMMENTED** [review]
This looks fine to me but I'll let @joelhawksley give the final ok.

**Comment by owenniblock** [comment]
@joelhawksley please let me know if I should make further changes before we can get this approved. If not, I'll remerge master today!

File: test/components/hellip_button_test.rb [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  def test_renders
+    render_inline(Primer::HellipButton.new("aria-label": "No effect"))
+
+    assert_selector("button[type='button'][aria-expanded='false']", text: "…")
+  end
+
+  def test_renders_inline
+    render_inline(Primer::HellipButton.new(inline: true, "aria-label": "No effect"))
+
+    assert_selector("button[type='button'][aria-expanded='false']", text: "…")
```
Comment (joelhawksley) [inline test/components/hellip_button_test.rb:17]: This assertion is the same as line 12. Shouldn't it be different?

Comment (owenniblock) [reply test/components/hellip_button_test.rb]: Good catch! I have changed it to include the `inline` class that's added.

**Review by joelhawksley — APPROVED** [review]

File: CHANGELOG.md [hunk 6]
```diff
@@ -36,12 +36,36 @@ The category for changes related to documentation, testing and tooling. Also, fo
 
     *Cameron Dutro*
 
+* Add the `HellipButton` component
+
+    *Amélia Chavot* & *Owen Niblock*
```
Comment (joelhawksley) [inline CHANGELOG.md:41]:
```suggestion
    *Amélia Chavot*, *Owen Niblock*
```

Comment (joelhawksley) [reply CHANGELOG.md:41]: (Same for below)

**Pull request merged**